easybuild/framework/easyblock.py

```diff
@@ -2209,6 +2209,9 @@ def fix_shebang(self):
                         if should_patch:
                             contents = shebang_regex.sub(shebang, contents)
                             write_file(path, contents)
+                        else:
```

Comment:
@Darkless012 This is not fully correct...

Imagine I have a bash script called `bin/foo` that has `#!/bin/bash`, and `fix_python_shebang_for = ['bin/*']` is used.

You don't want to inject `#!/usr/bin/env python` in that case and break the script.

We should make sure there's *no* shebang at all before adding it. Maybe like this:

```python
elif not contents.startswith('#!/'):

Reply:
Does the code path also verify that it is a script in the first place?? And not a random binary?

Reply:
@boegel Yup, I've also thought about having wrong files "fixed". Also, there is optional space possible after `#!` I will add that to tests as well.

Reply:
@akesandgren Yup, the files are read and checked for shebang regex or `#!` string. So binary files will not be touched. https://github.com/easybuilders/easybuild-framework/blob/1d112fb3aac5c62b58e80dc5bd15b2ebb1188513/easybuild/framework/easyblock.py#L2200-L2205

Reply:
@akesandgren Confirmed, that was fixed in #3029